lib/modules/datasource/endoflife-date/schema.ts

```diff
@@ -0,0 +1,41 @@
+import { DateTime } from 'luxon';
+import { z } from 'zod';
+import type { Release } from '../types';
+
+const EndoflifeDateVersionScheme = z
+  .object({
+    cycle: z.string(),
+    latest: z.optional(z.string()),
+    releaseDate: z.optional(z.string()),
+    eol: z.optional(z.union([z.string(), z.boolean()])),
+    discontinued: z.optional(z.union([z.string(), z.boolean()])),
+  })
+  .transform(({ cycle, latest, releaseDate, eol, discontinued }): Release => {
+    let isDeprecated = false;
+
+    // If "eol" date or "discontinued" date has passed or any of the values is explicitly true, set to deprecated
+    // "support" is not checked because support periods sometimes end before the EOL.
+    if (
+      eol === true ||
+      discontinued === true ||
+      (typeof eol === 'string' &&
+        DateTime.fromISO(eol, { zone: 'UTC' }) <= DateTime.now()) ||
```

Comment:
```suggestion
        DateTime.fromISO(eol, { zone: 'utc' }) <= DateTime.now().toUTC()) ||
```

otherwise you compare utc with local datetime

https://github.com/renovatebot/renovate/blob/0a4be4a989788fa7f920b1ff8bff3326970aea53/lib/modules/versioning/distro.ts#L134-L138